### DISCUSSION TITLE
Windows: Build packages not on C:/ drive

### DISCUSSION THREAD

**Discussion by foodwaterwifi** [microsoft/vcpkg#32582] [General]
Hi, vcpkg is a godsend of an app and saves much time setting up projects.

However, one gripe: I notice that vcpkg calls home the wonderful %APPDATA% directory of Windows, which is inseparably bound to the OS drive.

I do not have a very big OS drive (~128 GB, of which 5 GB remain), and heaven forbid I build LLVM and Clang from source on it intentionally or by mistake.

I am already struggling to windirstat this drive and delete the mess that many apps like to leave in these poor folders even after they are uninstalled. I am at the point where my efforts are met with diminishing returns.

I have a 2 TB D:/ drive and would much prefer to put everything on it instead. That includes vcpkg and anything it installs/builds.

I notice there are some options here to specify various folders for things: https://learn.microsoft.com/en-us/vcpkg/commands/common-options

It seems these must be called with each invocation, but that's difficult to do since I am using vcpkg with CMake and managing packages with vcpkg.json. Also, I would like to commit this vcpkg.json to a repo and not include hardcoded paths to my D:/ drive in it.

Is there a way to
1) Configure vcpkg so that it will never build/install packages on my C:/ drive in a way that is persistent?
2) ^

(it's fine if it has a minimal base of operations there, I just would not prefer big packages and binaries taking up space).

**Comment by autoantwort** [comment]
> It seems these must be called with each invocation, but that's difficult to do since I am using vcpkg with CMake and managing packages with vcpkg.json.

You can use https://learn.microsoft.com/en-us/vcpkg/users/buildsystems/cmake-integration#vcpkg_install_options

Also set the [binary cache](https://learn.microsoft.com/en-us/vcpkg/users/binarycaching) by setting the env var [`VCPKG_BINARY_SOURCES`](https://learn.microsoft.com/en-us/vcpkg/users/config-environment#vcpkg_binary_sources) to `clear;files,D:/vcpkg_binary_cache,readwrite`
